src/main/java/com/dev/theater/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.dev.theater.dao.impl;
+
+import com.dev.theater.dao.UserDao;
+import com.dev.theater.exception.CrudException;
+import com.dev.theater.library.Dao;
+import com.dev.theater.model.User;
+import com.dev.theater.util.HibernateUtil;
+import java.util.List;
+import java.util.Optional;
+import javax.persistence.criteria.CriteriaQuery;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new CrudException("Can't save " + user, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaQuery<User> criteriaQuery = session.getCriteriaBuilder()
```

Comment:
also, I would suggest to use HQL in all DaoImpl classes instead of Criteria API
we should better use the same approach everywhere